pkg/apis/aws/validation/infrastructure.go

```diff
@@ -65,6 +66,66 @@ func validateInfrastructureConfigZones(oldInfra, infra *apisaws.InfrastructureCo
 	return allErrs
 }
 
+// ValidateShootConfigAgainstCloudProfile validates the shoot with respect to the given (AWS) CloudProfile
+func ValidateShootConfigAgainstCloudProfile(shoot *core.Shoot, awsCloudProfile *aws.CloudProfileConfig, fldPath *field.Path) field.ErrorList {
```

Comment:
Anything related to "workers" better belong in the `apis/validation/shoot.go` - somewhere around https://github.com/gardener/gardener-extension-provider-aws/blob/806141bda213457d04cb950cbd1b968df18dd421/pkg/apis/aws/validation/shoot.go#L31

I would classify the image validation as part of the "worker" resource validation.